File: parse.py
import datetime
today = datetime.datetime.today()
year = today.strftime("%Y")
def parseDate(date):
    return datetime.datetime(int(year), *date).isoformat('T') + "Z"
def MakeTask(proposed_task):
    split_date = (int(x) for x in proposed_task["date"].split("/"))
    unslashed_content = proposed_task["content"].replace("\\", "")
    return {"title": unslashed_content, "due":parseDate(split_date)}







'''
DEPRECATED -- Using a basic algorithm to determine dates, obselete due to AI model


syllabus = open("syllabus.txt", "r").read().replace("\n"," newline ").split()
def parseDate(date):
    dates = date.split("/")
    if len(dates) > 1 and dates[0].isnumeric() and dates[1].isnumeric():
        return True
    return False
tasks = []
current_date = ""
current_label= ""
for i,word in enumerate(syllabus):
    if parseDate(word) or i == len(syllabus)-1:
        tasks.append({"date": current_date, "label":current_label})
        current_date = word
        current_label = ""
    elif word == "newline":
        current_label += "\n"
    else:
        current_label += word + " "
        
        '''

File: parse.js
const today = new Date();
const year = today.getFullYear();

function parseDate(date) {
  const [month, day] = date.split("/").map(Number); // Directly convert to numbers
  const newDate = new Date(year, month - 1, day); // Month is 0-indexed
    // Handle invalid dates
    if (isNaN(newDate.getTime())) {
        throw new Error("Invalid date format");  // Or handle it differently, e.g., return null
    }
  return newDate.toISOString();
}


function MakeTask(proposed_task) {
    const unslashed_content = proposed_task.content.replace(/\\/g, "");

    if (proposed_task.date) {
        try {
            console.log({ title: unslashed_content, due: parseDate(proposed_task.date) })
            return { title: unslashed_content, due: parseDate(proposed_task.date) };
        } catch (error) {
            console.error("Invalid date format:", proposed_task.date);
            // Fallback to returning the task without a due date
            return { title: unslashed_content }; 
        }
    } else {
        return { title: unslashed_content };
    }
}

module.exports = { MakeTask };
